### PR TITLE
[DOCS] Reformat script score query

### PR DESCRIPTION
**Pull request by jrodewig**
Updates the `script_score` query to use the new query format.

This creates separate sections for the example requests, parameters, and notes.

This is part of #40977, an effort to standardize documentation for query types.

### Preview
http://elasticsearch_45087.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/query-dsl-script-score-query.html

### CODE REVIEW

File: docs/reference/query-dsl/script-score-query.asciidoc [hunk 1]
```diff
@@ -4,18 +4,19 @@
 <titleabbrev>Script score</titleabbrev>
 ++++
 
-The `script_score` allows you to modify the score of documents that are
-retrieved by a query. This can be useful if, for example, a score
-function is computationally expensive and it is sufficient to compute
-the score on a filtered set of documents.
+Uses a <<modules-scripting,script>> to modify the <<relevance-scores,relevance
+scores>> of another query.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/script-score-query.asciidoc]: This is only one use case, but other use cases don't need to use relevance scores from another query. I would rephrase it something like this:

Use <<modules-scripting,script>> to provide/supply a custom score for documents.

File: docs/reference/query-dsl/script-score-query.asciidoc [hunk 2]
```diff
@@ -4,18 +4,19 @@
...
 
-To use `script_score`, you have to define a query and a script -
-a function to be used to compute a new score for each document returned
-by the query. For more information on scripting see
-<<modules-scripting, scripting documentation>>.
+The `script_score` query is useful if, for example, a scoring function is
+expensive and you only need to calculate the relevance score of a filtered set
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/script-score-query.asciidoc]: the relevance score -> score
Here a user can compute any score they like, it could be completely irrelevant to the IR relevance.

File: docs/reference/query-dsl/script-score-query.asciidoc [hunk 3]
```diff
@@ -4,18 +4,19 @@
...
+of documents.
 
-Here is an example of using `script_score` to assign each matched document
-a score equal to the number of likes divided by 10:
+
+[[script-score-query-ex-request]]
+==== Example request
+The following `script_score` query assigns each returned document a
+<<relevance-scores,relevance score>> equal to the `likes` field value divided by
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/script-score-query.asciidoc]: <relevance-scores,relevance score>>  -> score

File: docs/reference/query-dsl/script-score-query.asciidoc [hunk 4]
```diff
@@ -35,27 +36,58 @@ GET /_search
...
+[[script-score-top-level-params]]
+==== Top-level parameters for `script_score`
+`query`::
+(Required, query object) Query used to return documents.
+
+`script`::
++
+--
+(Required, <<modules-scripting-using,script object>>) Script used to modify
+the <<relevance-scores,relevance score>> of documents returned by the `query`.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/script-score-query.asciidoc]: Script used to modify the <<relevance-scores,relevance score>> -> Script used to compute the score of documents

File: docs/reference/query-dsl/script-score-query.asciidoc [hunk 5]
```diff
@@ -190,33 +222,44 @@ You can read more about decay functions
...
 <<vector-functions, Functions for vector fields>> are accessible through
 `script_score` query.
 
-==== Faster alternatives
-Script Score Query calculates the score for every hit (matching document).
-There are faster alternative query types that can efficiently skip
-non-competitive hits:
+[[script-score-faster-alt]]
+===== Faster alternatives
+The `script_score` query calculates the <<relevance-scores,relevance score>> for
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/script-score-query.asciidoc]: <<relevance-scores,relevance score>> - > score

File: docs/reference/query-dsl/script-score-query.asciidoc [hunk 6]
```diff
@@ -190,33 +222,44 @@ You can read more about decay functions
...
+the `script_score` query:
 
-Here we describe how Function Score Query's functions can be
-equivalently implemented in Script Score Query:
+* <<script-score>>
+* <<weight>>
+* <<random-score>>
+* <<field-value-factor>>
+* <<decay-functions>>
+* <<vector-functions>>
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/script-score-query.asciidoc]: <<vector-functions>> were not a part of the original `function_score` query, so we can exclude it from this list.

**Review by mayya-sharipova — APPROVED** [review]
@jrodewig Nice reworking! I left small changes to be completely precise in our docs, but otherwise LGTM

**Comment by jrodewig** [comment]
Thanks so much for the review and corrections/clarifications @mayya-sharipova!

File: docs/reference/query-dsl/script-score-query.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/query-dsl/script-score-query.asciidoc]: Fixed with a31f4a8ae60f8bbae76f211f53e1e88ee1a49541.

File: docs/reference/query-dsl/script-score-query.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/query-dsl/script-score-query.asciidoc]: Fixed with a31f4a8ae60f8bbae76f211f53e1e88ee1a49541.

File: docs/reference/query-dsl/script-score-query.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/query-dsl/script-score-query.asciidoc]: Fixed with 9028d83857f27699e62f7b7fa011b3f403059a43.

File: docs/reference/query-dsl/script-score-query.asciidoc (hunk 4 above)
Comment (jrodewig) [reply docs/reference/query-dsl/script-score-query.asciidoc]: Fixed with 9028d83857f27699e62f7b7fa011b3f403059a43.

File: docs/reference/query-dsl/script-score-query.asciidoc (hunk 5 above)
Comment (jrodewig) [reply docs/reference/query-dsl/script-score-query.asciidoc]: Fixed with d1750feaa1526c32a0a9ca5ba47bcc2b9e360615

File: docs/reference/query-dsl/script-score-query.asciidoc (hunk 6 above)
Comment (jrodewig) [reply docs/reference/query-dsl/script-score-query.asciidoc]: Fixed with c527ec1612ed47435fe6b6180d3443cfedffe3ad.
